src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -447,6 +447,40 @@ public unsafe void CopyTo(int sourceIndex, char[] destination, int destinationIn
                 elementCount: (uint)count);
         }
 
+        // TODO: https://github.com/dotnet/runtime/issues/51061
+        // Make these {Try}CopyTo methods public and use throughout dotnet/runtime.
+
+        /// <summary>Copies the contents of this string into the destination span.</summary>
+        /// <param name="destination">The span into which to copy this string's contents.</param>
+        /// <exception cref="System.ArgumentException">The destination span is shorter than the source string.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void CopyTo(Span<char> destination)
+        {
+            if ((uint)Length <= (uint)destination.Length)
+            {
+                Buffer.Memmove(ref destination._pointer.Value, ref _firstChar, (nuint)Length);
```

Comment:
```suggestion
                Buffer.Memmove(ref destination._pointer.Value, ref _firstChar, (uint)Length);
```

You'll get better codegen by explicitly casting int -\> uint before letting the compiler implicitly convert uint -\> nuint.

Reply:
It looks like the one impact is changing a movsxd to a mov, and if I'm reading the tables right, they appear to have the same cost. I changed it, anyway, but only for string right now. Does that have a material impact? Or maybe codegen has changed more for .NET 6 (I used sharplab which I believe is still 5).

Reply:
`mov` can be zero-latency, and it doesn't affect the register dependency chain like `movsxd` does. I did an experiment a while back where I changed the JIT to emit `mov` instead of `movsxd` in a naive "calculate the sum of all elements in this `int[]` array" benchmark and saw a ~50% gain in throughput.

Reply:
`mov reg, reg` _can_ be handled by the cpu-frontend with register renaming, so in that case it doesn't need to be actually executed (only fetched and decoded). So called zero-latency move. In order to make this happen, the necessary operands need to be ready (the case here) and the source and target register-sizes need to be compatible (the case here).

`movsxd` needs to be executed anyway (due the sign extension). So it's dispatched to the cpu-backend and uses an execution port.

@GrabYourPitchforks demonstrated this effect in a benchmark in https://github.com/dotnet/runtime/issues/12218#issue-557939556

> Or maybe codegen has changed more for .NET 6

AFAIK codegen didn't improve in that area.

Reply:
Got it. Thanks!